mirgecom/boundary.py

```diff
@@ -0,0 +1,92 @@
+__copyright__ = (
+    "Copyright (C) 2020 University of Illinois Board of Trustees"
+)
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import numpy as np
+import numpy.linalg as la  # noqa
+from pytools.obj_array import (
+    join_fields,
+    make_obj_array,
+    with_object_array_or_scalar,
+)
+import pyopencl.clmath as clmath
+from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
+
+# TODO: Remove grudge dependence?
+from grudge.eager import with_queue
+from grudge.symbolic.primitives import TracePair
+
+
+class DummyBoundary:
+    def __init__(self, tag=BTAG_ALL):
+        self._boundary_tag = tag
+        # dummy
+
+    def GetBoundaryFlux(self, discr, w, t=0.0):
+        ndim = discr.dim
+        dir_soln = discr.interp("vol", self._boundary_tag, w)
+
+        from mirgecom.euler import _facial_flux  # hrm
+
+        return _facial_flux(
+            discr,
+            w_tpair=TracePair(self._boundary_tag, dir_soln, dir_soln),
+        )
+
+
+class BoundaryBoss:
+    def __init__(self):
+        self._boundaries = {}
+
+    def AddBoundary(self, boundaryhandler):
+        numbnd = len(self._boundaries)
+        self._boundaries[numbnd] = boundaryhandler
+
+    def GetBoundaryFlux(self, discr, w, t=0.0):
```

Comment:
PEP8 naming. Here and elsewhere. (Flake8 in CI?)

Reply:
I've addressed some of this - removing as encountered.

Reply:
My point is more that there should be a Flake8 job in the CI.

[Example](https://github.com/inducer/grudge/blob/473f8ccb2b5b318064c39f9aca1c7d7d84ac49bd/.github/workflows/ci.yml#L13-L25)

Reply:
I think this is the subject of [this PR](https://github.com/illinois-ceesd/mirgecom/pull/10)?